prov/shm/src/smr_progress.c

```diff
@@ -222,20 +222,18 @@ static int smr_mmap_iov_copy_out(struct smr_ep *ep, struct smr_cmd *cmd,
 				 size_t *total_len)
 {
 	char shm_name[NAME_MAX];
-	char peer_name[NAME_MAX];
 	void *mapped_ptr;
 	int peer_id, fd, ret, num;
 
 	peer_id = (int) cmd->msg.hdr.addr;
 
-	sprintf(peer_name, "%s", ep->region->map->peers[peer_id].peer.name);
-	num = sprintf(shm_name, "%s_%s_%ld", peer_name, ep->name,
-		      cmd->msg.hdr.msg_id);
+	num = snprintf(shm_name, NAME_MAX - 1, "%s_%s_%ld",
+		     ep->region->map->peers[peer_id].peer.name, ep->name,
+		     cmd->msg.hdr.msg_id);
```

Comment:
Don't mix changes together in a single patch.

Reply:
Updated